src/thermo/include/antioch/nasa7_curve_fit.h

```diff
@@ -175,14 +175,19 @@ namespace Antioch
 
     for (unsigned int i=begin_interval; i != end_interval; ++i)
       {
-         const CoeffType *a = this->coefficients(interval);
+         const CoeffType *a = this->coefficients(i);
+
+         typedef typename Antioch::raw_value_type<StateType>::type raw_type;
 
          /* h/RT = a0     + a1*T/2 + a2*T^2/3 + a3*T^3/4 + a4*T^4/5 + a5/T */
         returnval = Antioch::if_else
         ( interval == i,
            StateType(  a[0] +
-                       a[1]*cache.T/2.0L + a[2]*cache.T2/3.0L + a[3]*cache.T3/4.0L +
-                       a[4]*cache.T4/5.0L + a[5]/cache.T),
+                       a[1]*cache.T/raw_type(2) +
```

Comment:
<pre>
./../../pbauman/src/thermo/include/antioch/nasa7_curve_fit.h:186:36: error: no match for ‘operator/’ (operand types are ‘const type {aka const vex::vector_expression<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::multiplies, boost::proto::argsns_::list2<vex::vector_expression<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<float>, 0l> >, const vex::vector<float>&>, 2l> >}’ and ‘long double’)
                        a[1]*cache.T/2.0L +

In file included from /Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/backend/opencl/source.hpp:41:0,
                 from /Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/backend/opencl.hpp:50,
                 from /Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/backend.hpp:74,
                 from /Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/vexcl.hpp:34,
                 from ../../../pbauman/test/vexcl_unit/nasa_mixture_vexcl_test.C:34:
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:75:25: note: candidate: cl_float2 operator/(const cl_float2&, const cl_float2&)
   inline cl_##name##len operator op(const cl_##name##len & a,                  \
                         ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:105:3: note: in expansion of macro ‘VEXCL_BIN_OP’
   VEXCL_BIN_OP(name, len, /)                                                   \
   ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:139:3: note: in expansion of macro ‘VEXCL_VEC_TYPE’
   VEXCL_VEC_TYPE(name, 2)                                                      \
   ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:158:1: note: in expansion of macro ‘VEXCL_TYPES’
 VEXCL_TYPES(float)
 ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:75:25: note:   no known conversion for argument 1 from ‘const type {aka const vex::vector_expression<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::multiplies, boost::proto::argsns_::list2<vex::vector_expression<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<float>, 0l> >, const vex::vector<float>&>, 2l> >}’ to ‘const cl_float2&’
   inline cl_##name##len operator op(const cl_##name##len & a,                  \
                         ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:105:3: note: in expansion of macro ‘VEXCL_BIN_OP’
   VEXCL_BIN_OP(name, len, /)                                                   \
   ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:139:3: note: in expansion of macro ‘VEXCL_VEC_TYPE’
   VEXCL_VEC_TYPE(name, 2)                                                      \
   ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:158:1: note: in expansion of macro ‘VEXCL_TYPES’
 VEXCL_TYPES(float)
 ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:90:25: note: candidate: cl_float2 operator/(const cl_float2&, const cl_float&)
   inline cl_##name##len operator op(const cl_##name##len & a,                  \
                         ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:109:3: note: in expansion of macro ‘VEXCL_BIN_SCALAR_OP’
   VEXCL_BIN_SCALAR_OP(name, len, /)                                            \
   ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:139:3: note: in expansion of macro ‘VEXCL_VEC_TYPE’
   VEXCL_VEC_TYPE(name, 2)                                                      \
   ^
/Users/pbauman/work/software/vexcl/vexcl-1.3.3-install/vexcl/types.hpp:158:1: note: in expansion of macro ‘VEXCL_TYPES’
 VEXCL_TYPES(float)
 ^
</pre>

etc. > 10000 lines of that.

I wonder if it's because we have `2.0L`, i.e. long double, to handle the cases where there's long double types in the non-VexCL cases, but the OpenCL only understands up to double to it's failing to convert to long double?